### PR TITLE
`azurerm_container_group` - set `storage_account_key` in the update operation

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description
API needs storage account key while updating the container group with storage account. Otherwise, the update API call would fail.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

Below failed test cases are also failed with same error on Teamcity Daily Run. So they are not related with this PR.
image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_container_group` - set `storage_account_key` in the update operation

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/26631

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Review by stephybun — COMMENTED** [review]
Thanks @neil-yechenwei.

Unfortunately all the tests for container group have been failing since the 2nd of July with the same error 
```
=== CONT  TestAccContainerGroup_AssignedIdentityUpdate
    testcase.go:121: Step 1/3 error: Error running apply: exit status 1
        Error: creating Container Group (Subscription: "*******"
        Resource Group Name: "acctestRG-240730232202481222"
        Container Group Name: "acctestcontainergroup-240730232202481222"): performing ContainerGroupsCreateOrUpdate: unexpected status 409 (409 Conflict) with error: RegistryErrorResponse: An error response is received from the docker registry 'index.docker.io'. Please retry later.
```

So I'm unable to run the tests for this. Would you be able to reach out to the service team about this?

**Comment by neil-yechenwei** [comment]
@stephybun , sure. I will check with service team about the failure.

**Comment by neil-yechenwei** [comment]
@stephybun , service team said that this is a known issue due to docker imposing stricter rate limits on docker public registry : [DockerHub Rate limit with Azure Container Instance and AKS | by alaa barqawi | Jul, 2024 | Medium](https://medium.com/@alaa.barqawi/docker-rate-limit-with-azure-container-instance-and-aks-4449cede66dd).  When a rate limit is hit, ACI is unable to pull images from docker, and deployment fails.
 To avoid these errors, we are requesting customers to use images from MCR or use ACR for their private images. So I updated this PR accordingly. Below is the test result I just now triggered. All related test cases are passed.

image

File: internal/services/containers/container_group_resource.go [hunk 1]
```diff
@@ -857,14 +857,41 @@ func resourceContainerGroupUpdate(d *pluginsdk.ResourceData, meta interface{}) e
...
 	}
+	if containerGroupVolumes != nil {
+		containerGroupVolumes = append(containerGroupVolumes, containerVolumes...)
+	}
+	model.Properties.Volumes = pointer.To(containerGroupVolumes)
 
+	// As Update API doesn't support to update identity, so it has to use CreateOrUpdate API to update identity
 	if err := client.ContainerGroupsCreateOrUpdateThenPoll(ctx, *id, model); err != nil {
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
```
Comment (stephybun) [inline internal/services/containers/container_group_resource.go:882]: If only the tags are updated then we would be making an unnecessary update call. I think this whole block should be moved into 
```
if d.HasChange("identity") {
...
}
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for fixing the tests @neil-yechenwei. I left one more suggestions in-line, once that's resolved this should be good to go.

File: internal/services/containers/container_group_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/containers/container_group_resource.go]: Updated

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comments. I updated PR. Please take another look. Below is the latest test result I just now triggered.
image
image

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM ⭐

**Pull request merged**